### PR TITLE
Huge website refactor to use React and cache notebook output

### PR DESCRIPTION
**Pull request by ry**
Known Issues:
- New cell button in notebook.html doesn't work.
- Shift-Enter doesn't change focus to the next cell.
- `notebook_mnist.html` was removed (this is ok - we can bring it back once memory management is done)
- syntax highlighting was removed for the "npm install" examples on `index.html` (this is okay for now)
- `notebook-cell-running` and `notebook-cell-updating` are not working.

### CODE REVIEW

File: nb_transpiler.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { assert } from "./util";
 
 const importFn = "__import";
 const globalVar = "__global";
+const consoleVar = "__console";
```
Comment (piscisaureus) [inline nb_transpiler.ts]: this variable is not used.

File: tools/build_website.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        const bodyHtml = document.body.innerHTML;
+        const html =  website.getHTML(page.title, bodyHtml);
+        resolve(html);
+      } catch (e) {
+        reject(e);
+      }
+    });
+  });
+  return p;
```
Comment (piscisaureus) [inline tools/build_website.ts]: return await p?

File: nb_transpiler.ts (hunk 1 above)
Comment (ry) [reply nb_transpiler.ts]: fixed

File: tools/build_website.ts (hunk 2 above)
Comment (ry) [reply tools/build_website.ts]: I think it's correct as is

**Comment by ry** [comment]
Will do

**Pull request merged**